app/screens/Reusable/Timers/CharmTimerButton/CharmTimerButton.tsx

```diff
@@ -0,0 +1,51 @@
+import _ from 'lodash';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { SafeAreaView } from 'react-native';
+
+import { IMAGES } from '../../../../assets';
+import {
+  BottomButtonView,
+  ExtendedButton,
+  ExtendedText,
+  MainHeader,
+  Timer,
+} from '../../../../components';
+import { generalStyles } from '../../../../utils/styles';
+import { styles } from './CharmTimerButton.styles';
+import { ICharmTimerButtonScreenProps } from './CharmTimerButton.types';
+
+export const CharmTimerButtonScreen: React.FC<ICharmTimerButtonScreenProps> = ({
+  navigation,
+  route,
+}) => {
+  const { t } = useTranslation();
+
+  const {
+    duration,
+    onSubmit = _.noop,
+    onButtonPress = _.noop,
+    subtitleKey = '',
+  } = route.params?.data;
```

Comment:
Destructing from `undefined` will lead to crash. Please, refactor this particular case.

Reply:
But these fields are required and without data these screens won't be used.